crates/telio-sockets/src/socket_pool.rs

```diff
@@ -320,4 +324,74 @@ mod tests {
 
         assert_eq!(socks.lock().unwrap().clone(), vec![tcp.as_native_socket()]);
     }
+
+    #[rstest]
+    #[case(IpAddr::V4(Ipv4Addr::LOCALHOST))]
+    #[cfg(not(windows))]
+    #[case(IpAddr::V4(Ipv4Addr::UNSPECIFIED))]
+    #[case(IpAddr::V6(Ipv6Addr::LOCALHOST))]
+    #[cfg(not(any(windows, tarpaulin)))]
+    #[case(IpAddr::V6(Ipv6Addr::UNSPECIFIED))]
+    #[tokio::test]
+    async fn internal_ipv6_udp_socket_can_transfer_data(#[case] ip_addr: IpAddr) {
```

Comment:
Does this only send on IPv6 socket?

Reply:
Good point, forgot to update the test name after I've added the ipv4 cases.